esp-hal/src/peripheral.rs

```diff
@@ -346,29 +344,21 @@ mod peripheral_macros {
             }
 
             impl $crate::private::Sealed for $name {}
-        };
-        ($(#[$cfg:meta])? $name:ident <= $base:ident) => {
-            $(#[$cfg])?
-            #[derive(Debug)]
-            #[allow(non_camel_case_types)]
-            /// Represents a concrete hardware peripheral.
-            ///
-            /// This struct is generated by the `create_peripheral!` macro when the peripheral
-            /// is tied to an actual hardware device.
-            pub struct $name { _inner: () }
 
-            $(#[$cfg])?
-            impl $name {
-                /// Unsafely create an instance of this peripheral out of thin air.
-                ///
-                /// # Safety
-                ///
-                /// You must ensure that you're only using one instance of this type at a time.
-                #[inline]
-                pub unsafe fn steal() -> Self {
-                    Self { _inner: () }
+            $(
+                impl $crate::dma::PeripheralMarker for $crate::peripherals::$name {
+                    #[inline(always)]
+                    fn peripheral(&self) -> $crate::system::Peripheral {
+                        $crate::system::Peripheral::$enum_variant
```

Comment:
```suggestion
                        paste::paste! { $crate::system::Peripheral::[< $name:camel >]}
```

Are we able to take advantage of `paste` here for case conversion, or does this not work reliably in this case?

Reply:
not every peripheral exists in the enum

Reply:
Ahh okay had misread how the macro works, no worries.

Reply:
I guess I can change the macro to mark these wih `[DMA-eligible]` and use `paste`, that might be cleaner maybe?

Reply:
Unfortunately `:camel` converts `I2S0` into `I2S0` so it's not suitable for us.